### PR TITLE
Edit tracks

### PR DESCRIPTION
**Pull request by naazy**
#37 editing/deleting tracks

### CODE REVIEW

File: src/MainUpdate.elm [hunk 1]
```diff
@@ -510,3 +511,52 @@ update msg model =
...
+            UpdatePickedTrack trackId trackInputType newPickedTrackInput ->
+                let
+                    pickedTrack =
+                        model.pickedTracks
+                            |> List.filter (\t -> t.id == trackId)
+                            |> List.head
+                            |> Maybe.withDefault (Track 0 "" "")
+
+                    newPickedTrack =
+                        if trackInputType == "name" then
```
Comment (roryc89) [inline src/MainUpdate.elm]: Instead of strings, we should use union types to handle this type of branching

Comment (naazy) [inline src/MainUpdate.elm]: Is that better than having 2 separate cases? One for updating track name and the other for updating track description?

File: src/MainUpdate.elm [hunk 2]
```diff
@@ -510,3 +511,52 @@ update msg model =
...
+                                | description = newPickedTrackInput
+                            }
+                        else
+                            pickedTrack
+
+                    pickedTracksWithoutUpdatedTrack =
+                        model.pickedTracks
+                            |> List.filter (\s -> s.id /= trackId)
+                in
+                    ( { model | pickedTracks = List.append pickedTracksWithoutUpdatedTrack [ newPickedTrack ] }, Cmd.none )
```
Comment (roryc89) [inline src/MainUpdate.elm]: Doesn't need to be changed but it's good to get into the habit of using cons `::` over append if possible as it runs in constant time

**Comment by roryc89** [comment]
Great stuff - my only suggestion is to use union types to specify logic branches.

File: src/MainUpdate.elm (hunk 1 above)
Comment (roryc89) [inline src/MainUpdate.elm]: you would still have 2 separate cases but they would be decided by what type the union type is.

Comment (roryc89) [inline src/MainUpdate.elm]: They're easier to follow for someone else reading the code, allow elm compiler to check our logic and prevent typo bugs

Comment (naazy) [inline src/MainUpdate.elm]: :+1:

Comment (naazy) [inline src/MainUpdate.elm]: Changed

File: src/MainModel.elm [hunk 3]
```diff
@@ -240,3 +242,8 @@ type alias ApiUpdateGet =
         List Column
     , submissions : List Submission
     }
+
+
+type TrackFields
+    = Name
```
Comment (roryc89) [inline src/MainModel.elm]: :+1:

**Pull request merged**